pkg/liqo-controller-manager/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -197,6 +191,16 @@ func (r *ForeignClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	tracer.Step("Ensured the ForeignCluster is processable")
 
+	// When the networking is not managed by liqo than the networking check is done before
+	// we do the other checks that could rely on networking.
+	if foreignCluster.Spec.NetworkingEnabled == discoveryv1alpha1.NetworkingEnabledNo {
+		// check for TunnelEndpoints
+		if err = r.checkTEP(ctx, &foreignCluster); err != nil {
+			klog.Error(err)
+			return ctrl.Result{}, err
+		}
+		tracer.Step("Checked the TunnelEndpoint status")
+	}
```

Comment:
Would personally avoid duplicating the tep status check, doing it always here regardless of the peering type (AFAIU it should cause no problems).